zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+                self.task.lastError = "image [uuid: %s] upload canceled" % cmd.imageUuid
+                shell.run('rbd rm %s' % task.tmpPath)
+
+        start = len(self.UPLOAD_PROTO)
+        imageUuid = cmd.url[start:start + self.LENGTH_OF_UUID]
+        dstPath = self._normalize_install_path(cmd.installPath)
+
+        pool, image_name = self._parse_install_path(cmd.installPath)
+        tmp_image_name = 'tmp-%s' % image_name
+        tmpPath = '%s/%s' % (pool, tmp_image_name)
+
+        task = UploadTask(imageUuid, cmd.installPath, dstPath, tmpPath)
```

Ruff (0.13.1)

`UploadTask` may be undefined, or defined from star imports

(F405)